homeassistant/components/geniushub/__init__.py

```diff
@@ -106,18 +118,102 @@
 )
 
 
-async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+async def validate_input(
```

Comment:
Why does this exist here and not in the config flow?

Reply:
Simply because I didn't want to spoof up the asyncio for testing.

Reply:
I don't understand what you mean here

Reply:
That should still happen. The requirement to have tests is because the config flow is something every user goes through and we need to be able to catch all things that fail. So we should not just mock this out.

If you have troubles with the mocking etc, I'll gladly help.

Reply:
I have been away for a few days. If you can point me at some examples, I can move the code and tests around.

I think, from memory, I was simply following another integration.

Reply:
There is no perfect example to look into. `epion` and `airgradient` are nice implementations. But be sure to check what you need before implemting something you dont need

Reply:
While it would be better to refactor the test, I am not going to rewrite the tests.